### PR TITLE
stdlib expansion and Demo prep

### PR DESCRIPTION
**Pull request by FuzzyNovaGoblin**
this includes the demo prep which was mostly expansion to the stdlib and then further expansion the the stdlib after the demo

### CODE REVIEW

File: src/fender_value.rs [hunk 1]
```diff
@@ -78,6 +79,64 @@ impl FenderValue {
         })
     }
 
+    pub fn shuffle(&mut self) -> Result<&mut FenderValue, String> {
```
Comment (Redempt) [inline src/fender_value.rs:82]: Remove shuffle from FenderValue

File: src/fender_value.rs [hunk 2]
```diff
@@ -78,6 +79,64 @@ impl FenderValue {
...
+            e => {
+                return Err(format!(
+                    "shuffle() can only be called on type `List`, expected `List` found `{:?}`",
+                    e.get_type_id()
+                ))
+            }
+        })
+    }
+
+    pub fn push(&mut self, value: FenderReference) -> Result<(), String> {
```
Comment (Redempt) [inline src/fender_value.rs:115]: Implement this for string as well

File: src/fender_value.rs [hunk 3]
```diff
@@ -78,6 +79,64 @@ impl FenderValue {
...
+                Ok(())
+            }
+            e => Err(format!(
+                "Can only call push on list: Expected type `List` found `{:?}`",
+                e.get_type_id()
+            )),
+        }
+    }
+
+    pub fn pop(&mut self) -> Result<FenderReference, String> {
```
Comment (Redempt) [inline src/fender_value.rs:129]: Implement for string

File: src/fender_value.rs [hunk 4]
```diff
@@ -99,6 +158,59 @@ impl FenderValue {
             v => v.clone(),
         }
     }
+
+    pub fn remove_at(&mut self, pos: i64) -> Result<FenderReference, String> {
```
Comment (Redempt) [inline src/fender_value.rs:162]: Implement for string

File: src/fender_value.rs [hunk 5]
```diff
@@ -99,6 +158,59 @@ impl FenderValue {
...
+            ));
+        }
+
+        Ok(list.remove(pos))
+    }
+}
+
+/// Cast functions
+impl FenderValue {
+    pub fn to_bool(&self) -> FenderValue {
```
Comment (Redempt) [inline src/fender_value.rs:199]: Change this to be a single function that takes the type to cast to

File: src/interpreter/error.rs [hunk 6]
```diff
@@ -2,17 +2,46 @@ use std::{error::Error, fmt::Display};
...
+    UnresolvedLabel(String, usize),
+}
+
+impl InterpreterError {
+    pub fn src_relative_string(&self, src: &str) -> String {
+        let (err_str, pos) = match self {
+            InterpreterError::UnresolvedName(name, pos) => {
+                (format!("Unresolved name `{name}`"), pos)
+            }
+            InterpreterError::DuplicateName(name, pos) => {
```
Comment (Redempt) [inline src/interpreter/error.rs:16]: Two of these error messages are incorrect

File: src/operators.rs [hunk 7]
```diff
@@ -117,10 +117,21 @@ fn eq(a: &FenderReference, b: &FenderReference) -> FenderReference {
     let b_val: &FenderValue = b.into();
     match (a_val, b_val) {
         (FenderValue::String(a), FenderValue::String(b)) => FenderValue::Bool(a == b).into(),
+        (FenderValue::Char(a), FenderValue::Char(b)) => FenderValue::Bool(a == b).into(),
         _ => num_eq(a, b),
     }
 }
 
+fn ne(a: &FenderReference, b: &FenderReference) -> FenderReference {
```
Comment (Redempt) [inline src/operators.rs:125]: Make this return false if the types are different

File: src/stdlib/control_flow.rs [hunk 8]
```diff
@@ -49,6 +49,8 @@ fndr_native_func!(
     then_func,
     |ctx, cond, body| {
         if matches!(*cond, FenderValue::Null | FenderValue::Error(_)) {
+            // dbg!("here");
```
Comment (Redempt) [inline src/stdlib/control_flow.rs:52]: Remove these comments

File: src/stdlib/control_flow.rs [hunk 9]
```diff
@@ -80,25 +82,60 @@ fndr_native_func!(
...
+                ctx.call(func, vec![incoming_val.get_pass_object()])?;
+            }
+            e => {
+                eprintln!(
+                    "\nAlso must take a function: Expected type `Function` found type `{}`",
+                    func.get_type_id().to_string()
+                );
+            }
+        }
+        // let Function(func) = &*func else{
```
Comment (Redempt) [inline src/stdlib/control_flow.rs:123]: Remove these comments

File: src/test/mod.rs [hunk 10]
```diff
@@ -155,3 +155,26 @@ fn run_quicksort_test() {
         )
     );
 }
+
+#[test]
+fn test_shuffle() {
+    // let input_list = (0..100).collect::<Vec<_>>();
```
Comment (Redempt) [inline src/test/mod.rs:161]: Fix this test

**Review by Redempt — CHANGES_REQUESTED** [review]

File: src/fender_value.rs (hunk 4 above)
Comment (FuzzyNovaGoblin) [reply src/fender_value.rs:162]: what would that return?

Comment (FuzzyNovaGoblin) [reply src/fender_value.rs:162]: what would that do on string?
(this msg may have been desynced for me)

Comment (Redempt) [reply src/fender_value.rs:171]: Remove a single character from the given index

Comment (FuzzyNovaGoblin) [reply src/fender_value.rs:171]: in vscode this comment showed up on a different function

**Pull request merged**